### PR TITLE
Feat/anonymized submodel data bruno tests

### LINKED ISSUE #1080: [Story] Bruno tests for anonymized delivery exchange
**As** PURIS admin,
**I want** bruno integration tests for anonymized deliveries,
**so that** I can test the full exchange of anonymized data.

## Hints / Details

- the tests should be added as a new folder to the test section `Test_02-EDC`
- the test should include the following sections:
  - contract setup
  - submodel descriptor retrieval (including DTR negotiation)
  - retrieval of submodel data (including Negotiation for submodel Asset with the EDC)

### Outcome

- anonymized delivery data is exchanged successfully

### Acceptance Criteria

- variables like the semantic id are added as environment variables where appropriate
- key tests verify the structure of returned data models
- all tests are tagged with the new tag `exchange-anonymized-data`

## Out of Scope

- edge cases (the focus is on the successful exchange scenario)
- test suite documentation is to be handled separately due to the large volume of changes

### LINKED ISSUE #1081: [Story] Bruno tests for anonymized item stock exchange
**As** PURIS admin,
**I want** bruno integration tests for anonymized stocks,
**so that** I can test the full exchange of anonymized data.

## Hints / Details

- the tests should be added as a new folder to the test section `Test_02-EDC`
- the test should include the following sections:
  - contract setup
  - subdmodel descriptor retrieval (including DTR negotiation)
  - retrieval of submodel data (including Negotiation for submodel Asset with the EDC)

### Outcome

- anonymized stock data is exchanged successfully

### Acceptance Criteria

- variables like the semantic id are added as environment variables where appropriate
- key tests verify the structure of returned data models
- all tests are tagged with the new tag `exchange-anonymized-data`

## Out of Scope

- edge cases (the focus is on the successful exchange scenario)
- test suite decoumentation is to be handled separately due to the large volume of changes

### LINKED ISSUE #1082: [Story] Bruno tests for anonymized production exchange
**As** PURIS admin,
**I want** bruno integration tests for anonymized production,
**so that** I can test the full exchange of anonymized data.

## Hints / Details

- the tests should be added as a new folder to the test section `Test_02-EDC`
- the test should include the following sections:
  - contract setup
  - subdmodel descriptor retrieval (including DTR negotiation)
  - retrieval of submodel data (including Negotiation for submodel Asset with the EDC)

### Outcome

- anonymized delivery data is exchanged successfully

### Acceptance Criteria

- variables like the semantic id are added as environment variables where appropriate
- key tests verify the structure of returned data models
- all tests are tagged with the new tag `exchange-anonymized-data`

## Out of Scope

- edge cases (the focus is on the successful exchange scenario)
- test suite decoumentation is to be handled separately due to the large volume of changes

### PR DESCRIPTION
**Pull request by jakoballgaier**
## Description
resolves #1080
resolves #1081 
resolves #1082

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] Copyright and license header are present on all affected files ([TRG 7.02](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02)
- [x] Documentation Notice are present on all affected files ([TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07))
- [x] If helm chart has been changed, the chart version has been bumped to either next major, minor or patch level (compared to released chart).
- [x] **Changelog** updated (`changelog.md`) with PR reference and brief summary.
- [x] **Frontend** version bumped, if needed (`frontend/package.json`, `frontend/package-lock.json`)
- [x] **Backend** version bumped, if needed (`backend/pom.xml`)
- [x] **Open API** specification updated, if controllers have been changed (use python script `scripts/generate_openapi_yaml.py` with running customer backend)

### CODE REVIEW

File: local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/02-Contract Negotiation/02-Retrieve Submodel Data/01-Query Catalog for Submodel.bru
Comment (ReneSchroederLJ) [inline local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/02-Contract Negotiation/02-Retrieve Submodel Data/01-Query Catalog for Submodel.bru:1]: For better readability in the test logs I would opt to split the logic in the script into pre-request and post-response.

the pre-request script should:
- parse the subModelData and index
- set the non catalog-related variables
- output a suitable message block to the console to identify the submodel currently being tested

the post-response script should handle:
- identifying the correct catalog offer
- setting the catalog related variables

For the log output something like this would be good:

```shell
------------------------------------------------------------------------------------------------------------------
Testing submodel data transfer 1/3
Submodel name:          DeliveryInformationAnonymized
Semantic id:            urn:samm:io.catenax.delivery_information_anonymized:1.0.0#DeliveryInformationAnonymized
Submodel endpoint:      http://supplier-data-plane:9285/api/public/860fb504-b884-4009-9313-c6fb6cdc776b/submodel
------------------------------------------------------------------------------------------------------------------
```

File: local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/02-Contract Negotiation/02-Retrieve Submodel Data/07-Terminate Transfer.bru [hunk 1]
```diff
@@ -29,8 +29,40 @@ body:json {
   }
 }
 
+script:post-response {
+  const jsonData = res.getBody()
+  bru.setVar("SUBMODEL_EDR_AUTH", jsonData.authorization)
```
Comment (ReneSchroederLJ) [inline local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/02-Contract Negotiation/02-Retrieve Submodel Data/07-Terminate Transfer.bru]: this does not seem to belong here.

**Review by ReneSchroederLJ — CHANGES_REQUESTED** [review]
Thank you for your work. Works fine for the most part I just left a suggestion for better readability of the logs.

**Review by ReneSchroederLJ — APPROVED** [review]
LGTM. Thank you for your contribution
